homeassistant/components/velux/config_flow.py

```diff
@@ -43,13 +54,54 @@ async def async_step_user(
                 errors["base"] = "unknown"
             else:
                 return self.async_create_entry(
-                    title=user_input[CONF_HOST],
+                    title=user_input[CONF_NAME] or user_input[CONF_HOST],
                     data=user_input,
                 )
 
-        data_schema = self.add_suggested_values_to_schema(DATA_SCHEMA, user_input)
+        data_schema = self.discovery_schema or self.add_suggested_values_to_schema(
+            DATA_SCHEMA, user_input
+        )
         return self.async_show_form(
             step_id="user",
             data_schema=data_schema,
             errors=errors,
         )
+
+    async def async_step_dhcp(
+        self, discovery_info: DhcpServiceInfo
+    ) -> ConfigFlowResult:
+        """Handle discovery by DHCP."""
+        return await self._async_handle_discovery(
+            {
+                CONF_HOST: discovery_info.ip,
+                CONF_MAC: format_mac(discovery_info.macaddress),
+                CONF_NAME: discovery_info.hostname.upper().replace("LAN_", ""),
```

Comment:
Also, why do we use name over mac?

Reply:
Name contains part of the mac and can also be used by zeroconf, where mac is not provided.

Reply:
What does zeroconf provide?

Reply:
2025-01-08 22:58:16.665 DEBUG (MainThread) [custom_components.velux] Discovered via Zeroconf with info: ZeroconfServiceInfo(ip_address=ZeroconfIPv4Address('192.168.178.76'), ip_addresses=[ZeroconfIPv4Address('192.168.178.76')], port=80, hostname='VELUX_KLF_LAN_5FCC.local.', type='_http._tcp.local.', name='VELUX_KLF_LAN_5FCC._http._tcp.local.', properties={'path': 'VELUX_KLF_LAN_5FCC.local'})

Reply:
Right. Let's also store the Mac address. We can use this to link to our device, allowing us to link them with the same device from other integrations

Reply:
But I agree we should not use it as unique id

Reply:
I haven't stored it seperately as it seems it's stored automatically as key via "discovery_keys":{"dhcp":[{"domain":"dhcp","key":"646184005fcc","version":1}]

Reply:
Oh right. But I rather not depend on that as we generally don't depend on that

Reply:
I modified the PR, the discovered data will be stored now.